File: 3770-largest-prime-from-consecutive-prime-sum/3770-largest-prime-from-consecutive-prime-sum.py
class Solution:
    def isPrime(self,n:int)->bool:
        if n<2:
            return False
        if n%2==0:
            return n==2
        i=3
        while i*i<=n:
            if n%i==0:
                return False
            i+=2
        return True        

    def largestPrime(self, n: int) -> int:
        sum=2
        res=0
        if n>=2:
            res=2
        x=3
        while sum+x<=n:
            if self.isPrime(x):
                sum+=x
                if self.isPrime(sum) and sum<=n:
                    res=sum
            x+=1

        return res
        

File: 3770-largest-prime-from-consecutive-prime-sum/3770-largest-prime-from-consecutive-prime-sum.java
class Solution {
    boolean isPrime(int n){
        if(n<2)return false;
        if(n%2==0)return false;
        for(int i=3;i*i<=n;i+=2){
            if(n%i==0)return false;
        }
        return true;
    }

    public int largestPrime(int n) {
        int sum=2;int res=2;
        //if(n>=2)res=2;
        for(int i=3;i+sum<=n;i++){
            if(isPrime(i)){
                sum+=i;
                if(sum<=n && isPrime(sum))res=sum;
            }
        }
        return res;
    }
}